syntax="proto3";
package clientServer;

service ClientServer{
    rpc register(registerRequest) returns (registerResponse);
    rpc unregister(unregisterRequest) returns (unregisterResponse);
    rpc login(loginRequest) returns (loginResponse);
    rpc logout(logoutRequest) returns (logoutResponse);
    rpc stayingAlive(stayRequest) returns (stayResponse); //user & password required
    rpc putFile(putFileRequest) returns (putFileResponse);
    rpc receivedFile(receivedFileRequest) returns (receivedFileResponse);
    rpc getFile(getFileRequest) returns (getFileResponse);
}

message registerRequest{
    string user=1;
    string password=2;
}

message registerResponse{
    int32 value=1;
    string response=2;

}
message unregisterRequest{
    string user=1;
    string password=2;
}
message unregisterResponse{
    int32 value=1;
    string response=2;
}
message loginRequest{
    string user=1;
    string password=2;
}
message loginResponse{
    int32 value=1;
    string response=2;
}
message logoutRequest{
    string user=1;
    string password=2;
    
}
message logoutResponse{
    int32 value=1;
    string response=2;
}

message stayRequest{
    string id=1;
    string username=2;
}
message stayResponse{
    int32 value=1;
    string response=2;
}

message putFileRequest{
    string username=1;
    string filename=2;
    int32 size=3;

}
message putFileResponse{
    int32 value=1;
    string ip1=2;
}

message receivedFileRequest{
    string node_id=1;
    string filename=2;
    int32 type=3;

}
message receivedFileResponse{
    int32 value=1;
    string id=2;


}

message getFileRequest{
    string filename=1;

}
message getFileResponse{
    int32 value=1;
    repeated string ips=2;
    string response=3;
}

